src/main/java/edu/harvard/iq/dataverse/ingest/plugin/spi/ServiceRegistry.java

```diff
@@ -107,46 +102,47 @@
  * com.mycompany.mypkg.MyService
  * </pre>
  *
- * <p> The service provider classes should be to be lightweight and
- * quick to load.  Implementations of these interfaces should avoid
- * complex dependencies on other classes and on native code. The usual
- * pattern for more complex services is to register a lightweight
- * proxy for the heavyweight service.
+ * <p>
+ * The service provider classes should be to be lightweight and quick to load.
+ * Implementations of these interfaces should avoid complex dependencies on
+ * other classes and on native code. The usual pattern for more complex services
+ * is to register a lightweight proxy for the heavyweight service.
  *
- * <p> An application may customize the contents of a registry as it
- * sees fit, so long as it has the appropriate runtime permission.
+ * <p>
+ * An application may customize the contents of a registry as it sees fit, so
+ * long as it has the appropriate runtime permission.
  *
- * <p> For more details on declaring service providers, and the JAR
- * format in general, see the <a
+ * <p>
+ * For more details on declaring service providers, and the JAR format in
+ * general, see the <a
  * href="../../../../technotes/guides/jar/jar.html">
  * JAR File Specification</a>.
  *
  * @see RegisterableService
  *
  */
-public class ServiceRegistry {
+public class ServiceRegistry <E>{
```

Comment:
I looked into the `spi` package (what feels like) a long time ago and wondered why this was copied (stolen?) from Java.

I haven't looked at it since and I don't know exactly if you got the generics right (when to use `E` instead of some concrete class I don't know right now), but in this case the space is in the wrong place. :)